test/github/test_logic.py

```diff
@@ -698,7 +698,55 @@ def test_no_rerun_stale_checks_on_unapproved_pull_request(
         )
         mock_rerequest_check_run.assert_not_called()
 
-    @patch("src.github.logic.SGTM_FEATURE__ALLOW_CHECK_RERUN_ON_APPROVAL", False)
+    @patch("src.github.logic.SGTM_FEATURE__CHECK_RERUN_ON_APPROVAL_ENABLED", True)
+    @patch("src.github.logic.SGTM_FEATURE__CHECK_RERUN_THRESHOLD_HOURS", 1)
+    @patch(
+        "src.github.logic.SGTM_FEATURE__CHECK_RERUN_BASE_REF_NAMES",
+        ["main"],
+    )
+    @patch("src.github.logic.SGTM_FEATURE__AUTOMERGE_ENABLED", True)
+    @patch.object(github_client, "merge_pull_request")
+    def test_rerun_stale_checks_on_automerge_pr(
+        self, mock_merge_pull_request, mock_rerequest_check_run
+    ):
+        check_run = build(builder.check_run().completed_at("2020-01-13T14:59:58Z"))
+        pull_request = build(
+            builder.pull_request()
+            .base_ref_name("main")
+            .commit(
+                builder.commit()
+                .status(Commit.BUILD_SUCCESSFUL)
+                .check_suites([builder.check_suite().check_runs([check_run])])
+            )
+            .review(
+                builder.review()
+                .submitted_at("2020-01-13T14:59:58Z")
+                .state(ReviewState.APPROVED)
+            )
+            .label(
+                builder.label().name(
+                    github_logic.AutomergeLabel.AFTER_TESTS_AND_APPROVAL.value
+                )
+            )
+            .merged(False)
+        )
+        self.assertTrue(
+            github_logic.maybe_rerun_stale_checks_on_approved_pull_request(pull_request)
+        )
+        mock_rerequest_check_run.assert_called_once_with(
```

Comment:
I'm a little surprised that we only expect to rerequest the check_run once. My understanding of the logic is:
1. a PR with a `merge after tests and approval` with stale tests gets approved
2. we re-run it via `rerun after approval`
3. we re-run it again via `autorerun on automerge`

Would you be able to walk through why #3 doesn't get hit?

Reply:
I think 3 would still get hit but only after line 746. At this point, we havent run any automerge logic yet.

Reply:
ooh I see. Is this behaviour something we want to retain? We'll be making 2 back-to-back rerun requests to Testville in this pathway which could introduce race conditions